subsys/net/ip/tcp.c

```diff
@@ -972,6 +972,11 @@ int net_tcp_send_pkt(struct net_pkt *pkt)
 	return net_send_data(pkt);
 }
 
+static void flush_queue(struct net_context *context)
+{
+	(void)net_tcp_send_data(context, NULL, NULL);
```

Comment:
Comment to this function says:

~~~
        /* For now, just send all queued data synchronously.  Need to
         * add window handling and retry/ACK logic.
         */
~~~

So, it still would be a workaround, not a solution, and could lead to silent data loss.